CustomeDB/CustomeDB.cpp

```diff
@@ -65,55 +74,154 @@ class LinkedList {
                     break;
                 }
                 cur = cur->next;
             }
             if (!found) {
-                cout << "value not found in linked iist.\n";
+                cout << "Price not found in linked iist.\n";
+            }
+        }
+    }
+
+    void swapNodes(Node* cur, Node* index) {
+        Item temp;
+        temp.id = cur->data.id;
+        temp.name = cur->data.name;
+        temp.price = cur->data.price;
+        cur->data.id = index->data.id;
+        cur->data.name = index->data.name;
+        cur->data.price = index->data.price;
+        index->data.id = temp.id;
+        index->data.name = temp.name;
+        index->data.price = temp.price;
+    }
+
+    void sortByNameAes() {
+        Node* cur = head;
+        Node* index = NULL;
+        if (head == NULL) {
+            cout << "Linked list empty, nothing to sort \n";
+        }
+        else {
+            while (cur != NULL) {
+                index = cur->next;
+                while (index != NULL) {
+                    if (cur->data.name > index->data.name) {
+                        swapNodes(cur, index);
+                    }
+                    index = index->next;
+                }
+                cur = cur->next;
             }
         }
     }
 
+    void sortByNameDec() {
+        Node* cur = head;
+        Node* index = NULL;
+        if (head == NULL) {
+            cout << "Linked list empty, nothing to sort \n";
+        }
+        else {
+            while (cur != NULL) {
+                index = cur->next;
+                while (index != NULL) {
+                    if (cur->data.name < index->data.name) {
+                        swapNodes(cur, index);
+                    }
+                    index = index->next;
+                }
+                cur = cur->next;
+            }
+        }
+    }
+
+    size_t listSize() {
+        Node* cur = head;
+        size_t counter = 0;
+        while (cur != NULL) {
+            cur = cur->next;
+            counter++;
+        }
+        return counter;
+    }
+
+    void reverseLinkedList() {
+        Node* cur = head;
+        Node* temp = NULL;
+        Node* prev = NULL;
+        if (head == NULL) {
+            cout << "Linked list empty, nothing to reverse.\n";
+        }
+        else if (head->next == NULL) {
+            cout << "Only one element in List, nothing to reverse.\n";
+        }
+        else {
+            while (cur != NULL) {
+                temp = cur->next;
+                cur->next = prev;
+                prev = cur;
+                cur = temp;
+            }
+            head = prev;
+        }
+    }
+
 };
 
 int main()
 {
     LinkedList l1;
-    l1.addAtEnd(5);
-    l1.addAtEnd(70);
-    l1.deleteByValue(5);
+    l1.addAtEnd(1, "talha", 20);
+    l1.addAtEnd(2, "haseed", 10);
+    l1.addAtEnd(3, "bats", 25);
+    l1.displayList();
+    
+    l1.sortByNameAes();
+    l1.displayList();
+    l1.sortByNameDec();
     l1.displayList();
 
```

Comment:
there is memory leakage in your program. i want u to find it out. for hint run add data and delete at head hundreds of thousands of time try testing it.

Reply:
fixed.